### DISCUSSION TITLE
Docker v26.1.5 release?

### DISCUSSION THREAD

**Discussion by staal0** [moby/moby#48292] [Q&A] [answered]
### Description

Has Docker v26.1.5 not been released to repos?

### Reproduce

apt-cache madison docker-ce-cli | grep 26.1
docker-ce-cli | 5:26.1.4-1~ubuntu.20.04~focal | https://download.docker.com/linux/ubuntu focal/stable amd64 Packages
docker-ce-cli | 5:26.1.3-1~ubuntu.20.04~focal | https://download.docker.com/linux/ubuntu focal/stable amd64 Packages
docker-ce-cli | 5:26.1.2-1~ubuntu.20.04~focal | https://download.docker.com/linux/ubuntu focal/stable amd64 Packages
docker-ce-cli | 5:26.1.1-1~ubuntu.20.04~focal | https://download.docker.com/linux/ubuntu focal/stable amd64 Packages
docker-ce-cli | 5:26.1.0-1~ubuntu.20.04~focal | https://download.docker.com/linux/ubuntu focal/stable amd64 Packages

### docker version

```bash
26.1.5
```

### docker info

```bash
n/a
```

**Comment by thaJeztah** [comment]
No there won't be packages for that tag on download.docker.com; [project/BRANCHES-AND-TAGS.md](https://github.com/moby/moby/blob/265f0a7fa7d3bb824ac676dcfeb25ccaec670efe/project/BRANCHES-AND-TAGS.md#branches)

Tags in this repository are source releases, and can be used by packagers to build from; docker only maintains packages fro the current release (v27.x at time of writing), and does not currently provide LTS versions for older releases of docker-ce.

**Reply by jdonovan-itcnp** [reply]
One of my customers in the public sector is requesting that either this update be released to the repository, or that additional guidance be provided for how to resolve associated security issues. 

It appears that 26.1 is the latest version supported for CentOS 8 distributions in the docker repository, so we do not have a simple path to upgrade to 27. Is the intention to cease support for 26.1 and CentOS 8? It seems strange not to release a security-related patch.

**Reply by vvoland** [reply]
CentOS Stream 8 is EOL already in the upstream https://blog.centos.org/2023/04/end-dates-are-coming-for-centos-stream-8-and-centos-linux-7/

**Comment by firecow** [comment]
@thaJeztah Tags in this repository have been available up until the latest few patches on my machine

Seen here
```
~$ sudo apt-cache madison docker-ce
 docker-ce | 5:27.1.1-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:27.1.0-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:27.0.3-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:27.0.2-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:27.0.1-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:26.1.4-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:26.1.3-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:26.1.2-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:26.1.1-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:26.1.0-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:26.0.2-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:26.0.1-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:26.0.0-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:25.0.5-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:25.0.4-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:25.0.3-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:25.0.2-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:25.0.1-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:25.0.0-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:24.0.9-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:24.0.8-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:24.0.7-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:24.0.6-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:24.0.5-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:24.0.4-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:24.0.3-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:24.0.2-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:24.0.1-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:24.0.0-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:23.0.6-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:23.0.5-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:23.0.4-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:23.0.3-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:23.0.2-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:23.0.1-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:23.0.0-1~ubuntu.22.04~jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:20.10.24~3-0~ubuntu-jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:20.10.23~3-0~ubuntu-jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:20.10.22~3-0~ubuntu-jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:20.10.21~3-0~ubuntu-jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:20.10.20~3-0~ubuntu-jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:20.10.19~3-0~ubuntu-jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:20.10.18~3-0~ubuntu-jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:20.10.17~3-0~ubuntu-jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:20.10.16~3-0~ubuntu-jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:20.10.15~3-0~ubuntu-jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:20.10.14~3-0~ubuntu-jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
 docker-ce | 5:20.10.13~3-0~ubuntu-jammy | https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages
```

```
# cat /etc/apt/sources.list.d/docker.list
deb [arch=amd64 signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/ubuntu jammy stable
```

Something must have changed in the way you publish packages to download.docker.com.

As i read this 26.x.x will receive security patches. 
![image](https://github.com/user-attachments/assets/ff8e88a3-5464-4a53-a4ae-5cf262720480)

Isn't it pretty bad for security not providing customers/developers with security patches for major versions that aren't that old?

**Reply by kextrac** [reply]
I too am confused by the response that 26.1 is not maintained, and that binary release packages for a major version that is only one version older will not be released for a vulnerability with CVSS score of 9.9.  I cannot clearly identify if support policy really spells out the versions that would be in support scope, let alone anything about LTS.  I'll keep digging through docs...

**Reply by kextrac** [reply]
Browsing the tree various os distributions and versions, it seems that centos 8 has not been updated either for a long while either.  Not sure why?  Is centos deprecated (likely)?  Perhaps it would be helpful to document which OS/distribution is supported for X number of versions, etc.  Perhaps there is document, but I cannot find.

**Reply by AkihiroSuda** [reply]
> CVSS score of 9.9

The CVSS score of https://nvd.nist.gov/vuln/detail/CVE-2024-41110 is kinda misleading, as the vuln is not exploitable on most environments. (Almost nobody uses AuthZ plugins)

**Reply by madrover** [reply]
@cpuguy83 , it is stated at [project/BRANCHES-AND-TAGS.md](https://github.com/moby/moby/blob/265f0a7fa7d3bb824ac676dcfeb25ccaec670efe/project/BRANCHES-AND-TAGS.md#branches) that you're the current maintainer of the 26.1 branch for security issues. Is that correct? Should we expect a version 26.1.5 any time soon?
Thanks!

**Reply by thaJeztah** [reply, marked as answer]
The 26.1 branch is used as an upstream for Microsoft / Azure's builds of packages from the Moby repository. Docker no longer packages 26.x; if you're using packages from download.docker.com, it's best to upgrade to 27.x.

But as mentioned above, the CVE for 26.1.4 only impacts a very marginal group of users; if you're not using Authz plugins, you should not be affected.
